test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+	}
+	time.Sleep(time.Second)
+	return nil
+}
+
+func (s *systemtestSuite) SetupInfraNetwork() error {
+	err := s.cli.NetworkPost(&client.Network{
+		TenantName:  "default",
+		NetworkName: "default-net",
+		Subnet:      "100.10.1.0/24",
+		Encap:       "vxlan",
+	})
+	if err != nil {
+		logrus.Errorf("default-net not created. Err: %+v", err)
+		return err
+	}
+	time.Sleep(time.Second)
+	return nil
+}
+
+func (k *kubePod) newContainer(node *node, containerID, name string, spec containerSpec) (*container, error) {
+	cont := &container{
+		node:        node,
+		containerID: containerID,
+		name:        name,
+	}
+
+	out, err := k8sMaster.exec.getIPAddr(cont, "eth0")
+
+	if err != nil {
+		return nil, err
+	}
+	cont.eth0.ip = out
+
+	out, err = cont.node.exec.getIPv6Addr(cont, "eth0")
+	if err == nil {
+		cont.eth0.ipv6 = out
+	}
+
+	return cont, nil
+}
+
+func (k *kubePod) runContainer(spec containerSpec) (*container, error) {
+	var namestr, labelstr, image string
+	labels := []string{}
+
+	if len(spec.tenantName) != 0 && spec.tenantName != "default" {
+		labels = append(labels, "io.contiv.tenant="+spec.tenantName)
+	}
+
+	if spec.serviceName != "" {
+		labels = append(labels, "io.contiv.net-group="+spec.serviceName)
+	}
+	if spec.networkName != "" {
+		labels = append(labels, "io.contiv.network="+spec.networkName)
+	}
+
+	labelstr = strings.Join(labels, ",")
+
+	if len(labelstr) != 0 {
+		labelstr = "--labels=" + labelstr
+	}
+
+	image = "--image=contiv/alpine "
+
+	cmdStr := " --command -- sleep 900000"
+
+	if spec.name != "" {
+		namestr = spec.name
+	}
+
+	if len(spec.labels) > 0 {
+		l := " --labels="
+		for _, label := range spec.labels {
+			labelstr += l + label + " "
+		}
+	}
+
+	cmd := fmt.Sprintf("kubectl run %s %s %s --restart=Never %s ", namestr, labelstr, image, cmdStr)
+
+	logrus.Infof("Starting Pod %s on with: %s", spec.name, cmd)
+	out, err := k8sMaster.tbnode.RunCommandWithOutput(cmd)
+	if err != nil {
+		logrus.Errorf("cmd %q failed: output below", cmd)
+		logrus.Println(out)
+		return nil, err
+	}
+
+	//find out the node where pod is deployed
+
+	for i := 0; i < 50; i++ {
+		time.Sleep(5 * time.Second)
+		cmd = fmt.Sprintf("kubectl get pods -o wide | grep %s", spec.name)
+		////master.lock()
+		out, err = k8sMaster.tbnode.RunCommandWithOutput(cmd)
+		if strings.Contains(out, "Running") {
+			break
+		}
+	}
+
+	podInfoStr := strings.TrimSpace(out)
+
+	if podInfoStr == "" {
+		logrus.Errorf("Error Scheduling the pod")
+		return nil, errors.New("Error Scheduling the pod")
+	}
+
+	podInfo := strings.Split(podInfoStr, " ")
+
+	podID := podInfo[0]
+	nodeID := podInfo[len(podInfo)-1]
+
+	podNode := k.node.suite.vagrant.GetNode(nodeID)
+
+	n := &node{
+		tbnode: podNode,
+		suite:  k.node.suite,
+		exec:   k,
+	}
+
+	cont, err := k.newContainer(n, podID, spec.name, spec)
+	if err != nil {
+		logrus.Info(err)
+		return nil, err
+	}
+
+	return cont, nil
+}
+
+func (k *kubePod) checkPingFailure(c *container, ipaddr string) error {
```

Comment:
These are called from the actual system-tests. These functions are just an implementation of the abstracted functions that the systemtests call.